### PR TITLE
fix(annotations): Fix AnnotationControlsFSM reset when fullscreen enter

### PR DESCRIPTION
**Pull request by mxiao6**

### CODE REVIEW

File: src/lib/viewers/BaseViewer.js [hunk 1]
```diff
@@ -575,8 +575,8 @@ class BaseViewer extends EventEmitter {
 
         if (this.annotator && this.areNewAnnotationsEnabled()) {
             this.annotator.emit(ANNOTATOR_EVENT.setVisibility, false);
-            this.annotator.toggleAnnotationMode(AnnotationMode.NONE);
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:578]: Is this not still needed to exit "create" mode in the annotator?

File: src/lib/viewers/BaseViewer.js [hunk 2]
```diff
@@ -575,8 +575,8 @@ class BaseViewer extends EventEmitter {
...
             this.disableAnnotationControls();
+            this.processAnnotationModeChange(this.annotationControlsFSM.transition(AnnotationInput.RESET));
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:579]: Should this be in `disableAnnotationControls`, instead?

Comment (mxiao6) [reply src/lib/viewers/BaseViewer.js:579]: Yeah, you're right

File: src/lib/viewers/BaseViewer.js (hunk 1 above)
Comment (mxiao6) [reply src/lib/viewers/BaseViewer.js:578]: It's already in `disableAnnotationControls`

File: src/lib/viewers/BaseViewer.js [hunk 3]
```diff
@@ -919,8 +918,8 @@ class BaseViewer extends EventEmitter {
     disableAnnotationControls() {
         if (this.annotator && this.annotationControls && this.areNewAnnotationsEnabled()) {
             this.annotator.toggleAnnotationMode(AnnotationMode.NONE);
-            this.annotationControls.resetControls();
             this.annotationControls.toggle(false);
+            this.processAnnotationModeChange(this.annotationControlsFSM.transition(AnnotationInput.RESET));
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:922]: Can we move this outside of the conditional above? We'll need it for the react rewrite, which doesn't set `this.annotationControls`.

**Review by jstoffan — APPROVED** [review]

**Pull request merged**